plugins/gradle/gradle-plugin/src/main/java/org/wildfly/swarm/plugin/gradle/AbstractThorntailPlugin.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2018 Red Hat, Inc, and individual contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wildfly.swarm.plugin.gradle;
+
+import javax.inject.Inject;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.tooling.provider.model.ToolingModelBuilder;
+import org.gradle.tooling.provider.model.ToolingModelBuilderRegistry;
+
+/**
+ * The base plugin that initializes the tooling model.
+ */
+@SuppressWarnings("UnstableApiUsage")
+public class AbstractThorntailPlugin implements Plugin<Project> {
```

Comment:
I moved code from the `PackagePlugin` class in to an abstract class that is extended by the `PackagePlugin` as well as the `ThorntailArquillian` plugin. We need a separate vector (in this case a plugin) for targeting test cases. If we overload the existing plugin and insist that developers need to add it to their projects, then we will end up messing up the lifecycle of "library-modules" which don't have to be JEE applications.